m3/thriftudp/transport.go

```diff
@@ -122,6 +131,20 @@ func (p *TUDPTransport) Read(buf []byte) (int, error) {
 	return n, thrift.NewTTransportExceptionFromError(err)
 }
 
+// ReadByte reads a single byte and returns it
+func (p *TUDPTransport) ReadByte() (byte, error) {
+	n, err := p.Read(p.readByteBuf)
+	if n == 0 {
```

Comment:
Updated.

Just as a FYI, this case doesn't handle some of the intricacies of the `io.Reader`:
 - `Read` may return data *and* an error, which is valid, but `ReadByte` cannot do that
 - `Read` may also return 0 bytes read and no error

There's not a great way to handle these cases without making the `TUDPTransport` carry a buffer, which then the `Read` method would also need to handle etc. Since this method isn't used by production code (only tests) so I think erring on the side of returning an `err` and not handling the data makes sense and keeps it simple.